daml-foundations/daml-ghc/damldoc/src/DA/Daml/GHC/Damldoc/HaddockParse.hs

```diff
@@ -109,37 +116,61 @@ collectDocs = go Nothing []
     go Nothing docs (d:ds) = go (Just d) docs ds
     go (Just prev) docs (d:ds) = finished prev docs (go (Just d) [] ds)
 
-    finished decl docs rest = (unLoc decl, reverse docs) : rest
+    finished decl docs rest = (decl, toDocText . map unDocText . reverse $ docs) : rest
+
+-- | Context in which to extract a module's docs. This is created from
+-- 'TypecheckedModule' by 'buildDocCtx'.
+data DocCtx = DocCtx
+    { dc_mod :: Modulename
```

Comment:
`DocCtx` seems slightly weird since the `dc_tcmod` field basically seems to make all other fields redundant. Perfectly happy to have this as an intermediate state but maybe add a comment on what you expect the final state to look like.

Reply:
Yep ... this certainly isn't finished yet. Ultimately DocCtx will still be derivable from TypecheckedModule, but I'm thinking a little bit ahead to when I will probably add a map from function name / field name to type sig, so we can use it while processing function / field docs later.